04-Multi-Band-Raster.Rmd

```diff
@@ -261,28 +266,36 @@ RGB_stack@layers
 #view attributes for one layer
 RGB_stack[[1]]
 
+#view histogram of all 3 layers
+hist(RGB_stack)
+
 #plot one band
 plot(RGB_stack[[1]], 
-     main="band one", 
+     main="RGB_stack Band 1", 
      col=grayscale_colors)
 
-#plot all three bands
+#plot all three bands seperately
 plot(RGB_stack, 
      col=grayscale_colors)
 
+par(mfrow=c(1,1)) # go back 1 plot at time, not 2x2.
 ```
 
-#Plot A Three Band Image
 
-We can use `plotRGB` to compositive a 3 band color raster stack into a final 
-color image. This function allows us to also
+#Creating A Three Band Image
+As you just saw we can't use `plot()` to combine all three bands and create a 
+color image.  We can, however, use `plotRGB()` to composite a 3 band color
+raster stack into a final color image. 
 
-1. Determine what order we want to plot the bands (We could create a color infrared
-image using this function if we had an infrared band to work with)  
-2. Adjust the stretch of the image to make it brighter / darker
+This function allows us to also
 
-The plotRGB function composits three bands together into one producing an image
-that is similar to an image that a camera takes. 
+1. Determine what order we want to plot the bands.  RGB assumes that the bands 
+are 1=red, 2=green, and 3=blue.  However, bands may not be in that order or bands
+may be of data that is outside the visible human range yet we want to plot them 
+in RGB so that we can visualize it. Images from infrared cameras do this with the
+infrared band usually plotted as band 1, which appears red to us.   
+2. Adjust the `stretch` of the image to make it brighter or darker. Stretch 
```

Comment:
hmmm not really

Stretch FUNCTIONS on the historgram. It looks at the distribution of the values and then tells R how to assign the colors in the color ramp across all values.  so, for example linear might clip the tailes of the data and make all values in the tales either black or white. i'm not sure how to expain this.

Maybe we can use the graphic on this page - 

http://www.nrcan.gc.ca/earth-sciences/geomatics/satellite-imagery-air-photos/satellite-imagery-products/educational-resources/9389

we can also ask colin to make some graphics!